dvc/repo/diff.py

```diff
@@ -156,8 +156,13 @@ def onerror(target, _exc):
         for target in old_missing & new_missing:
             raise FileNotFoundError(errno.ENOENT, os.strerror(errno.ENOENT), target)
 
-    old = indexes[a_rev]
-    new = indexes[b_rev]
+    if len(indexes.keys()) == 1:
```

Comment:
The fact that brancher joins rev names feels wrong, should do something about it (maybe return both name and sha so that the user could concat them if he pleases so).

Reply:
Let's run with this for now, brancher refactor needs more time.